app/views/RoomView/ReactionPicker.tsx

```diff
@@ -1,85 +1,99 @@
 import React from 'react';
 import { View } from 'react-native';
-import { connect } from 'react-redux';
-import Modal from 'react-native-modal';
+import { Q } from '@nozbe/watermelondb';
 
 import EmojiPicker from '../../containers/EmojiPicker';
-import { isAndroid } from '../../lib/methods/helpers';
-import { themes } from '../../lib/constants';
-import { TSupportedThemes, withTheme } from '../../theme';
+import { useTheme } from '../../theme';
 import styles from './styles';
-import { IApplicationState } from '../../definitions';
-
-const margin = isAndroid ? 40 : 20;
-const maxSize = 400;
+import { IApplicationState, IEmoji } from '../../definitions';
+import { EventTypes } from '../../containers/EmojiPicker/interfaces';
+import { useAppSelector } from '../../lib/hooks';
+import { FormTextInput } from '../../containers/TextInput/FormTextInput';
+import I18n from '../../i18n';
+import { sanitizeLikeString } from '../../lib/database/utils';
+import { emojis } from '../../containers/EmojiPicker/emojis';
+import database from '../../lib/database';
+import { debounce } from '../../lib/methods/helpers/debounce';
 
 interface IReactionPickerProps {
-	baseUrl: string;
 	message?: any;
 	show: boolean;
-	isMasterDetail: boolean;
 	reactionClose: () => void;
 	onEmojiSelected: (shortname: string, id: string) => void;
 	width: number;
 	height: number;
-	theme: TSupportedThemes;
 }
 
-class ReactionPicker extends React.Component<IReactionPickerProps> {
-	shouldComponentUpdate(nextProps: IReactionPickerProps) {
-		const { show, width, height } = this.props;
-		return nextProps.show !== show || width !== nextProps.width || height !== nextProps.height;
-	}
+const MAX_EMOJIS_TO_DISPLAY = 20;
+
+const ReactionPicker = React.memo(({ onEmojiSelected, message, reactionClose }: IReactionPickerProps) => {
```

Comment:
This react.memo will not work properly, even more so since message is a giant object.